pgx/src/bgworkers.rs

```diff
@@ -210,10 +212,130 @@ unsafe extern "C" fn worker_spi_sigterm(_signal_args: i32) {
     pg_sys::SetLatch(pg_sys::MyLatch);
 }
 
+/// Dynamic background worker handle
+pub struct BackgroundWorkerHandle {
+    handle: *mut pg_sys::BackgroundWorkerHandle,
+    notify_pid: pg_sys::pid_t,
+}
+
+/// PID
+pub type Pid = pg_sys::pid_t;
+
+/// Dynamic background worker status
+#[derive(Debug, Clone, Copy)]
+pub enum BackgroundWorkerStatus {
+    Started,
+    NotYetStarted,
+    Stopped,
+    PostmasterDied,
+    /// `BackgroundWorkerBuilder.bgw_notify_pid` was not set to `pg_sys::MyProcPid`
+    ///
+    /// This makes worker's startup or shutdown untrackable by the current process.
+    Untracked {
+        /// `bgw_notify_pid` as specified in the builder
+        notify_pid: pg_sys::pid_t,
+    },
+}
+
+impl From<pg_sys::BgwHandleStatus> for BackgroundWorkerStatus {
+    fn from(s: pg_sys::BgwHandleStatus) -> Self {
+        match s {
+            pg_sys::BgwHandleStatus_BGWH_STARTED => BackgroundWorkerStatus::Started,
+            pg_sys::BgwHandleStatus_BGWH_NOT_YET_STARTED => BackgroundWorkerStatus::NotYetStarted,
+            pg_sys::BgwHandleStatus_BGWH_STOPPED => BackgroundWorkerStatus::Stopped,
+            pg_sys::BgwHandleStatus_BGWH_POSTMASTER_DIED => BackgroundWorkerStatus::PostmasterDied,
+            _ => unreachable!(),
+        }
+    }
+}
```

Comment:
I should note this is not a blocking concern, or even necessarily a design I would like more, it's just an implicit decision that is being made here that I felt I should note, and ideally we would document why we made the decision one way or another. I made it slightly differently when writing `impl TryFrom<c_int> for Spi{Ok,Error}`, but that case may be sufficiently different from this one to justify different decisions (likely most relative to probable evolution).

https://github.com/tcdi/pgx/blob/b92c60974c203e07e05cf1320581084be20a8e8a/pgx/src/spi.rs

Reply:
Ahh re-reviewing the commits I spotted the one where you removed the Unknown variant as useless. Like I said, I think it's fine to do whatever as long as we explain to our future selves why we thought it was a good idea at the time.

Reply:
I think the idea in the earlier discussion was that pgx effectively gate-keeps the changes to variants, and we know that variants won't change.

I am happy to revisit this whole thing as a follow-up PR, just want to unblock this one so that we have a smaller surface to iterate on.

Reply:
Yeah we can revisit this a bit later then.